frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx

```diff
@@ -16,7 +16,14 @@ interface TopologyControlBarProps {
 }
 
 const TopologyControlBar: React.FC<TopologyControlBarProps> = ({ visualization }) => {
-  const layout = visualization.getGraph()?.getLayout() ?? 'COLA_LAYOUT';
+  const [layout, setLayout] = React.useState<string>(
+    visualization?.getGraph()?.getLayout() ?? COLA_LAYOUT,
+  );
+  React.useEffect(() => {
+    visualization.getGraph().setLayout(layout);
+    visualization.getGraph().layout();
+  }, [layout, visualization]);
```

Comment:
Sorry that I missed this in the first round.. This effect was also called after the first rendering and it calls setLayout and layout right after the initial rendering. When seeing this I checked that setLayout [do nothing if the layoutType does not changed](https://github.com/patternfly/patternfly-react/blob/2c4b25ce8aa64a08de38c0a1884c96df643f348e/packages/react-topology/src/elements/BaseGraph.ts#L109-L111).

If the initial layout call here required? Should we ignore this or do we want do this only when the user pressed one of the buttons? :thinking:

Reply:
Ya we should avoid a call to `layout` here on first render.

Alternative to `useState` would be to just make the control bar an `observer` which will force re-render when layout changes using the old implementation.

Reply:
Or useState/useRef and call setLayout and layout again in an onClick handler.

Reply:
Thanks. Just added the observer and updated to use the constant.